packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2837,19 +2826,19 @@ export class ObjectsInRedisClient {
                                 CONSTS.ACCESS_USER_RW | CONSTS.ACCESS_GROUP_READ | CONSTS.ACCESS_EVERY_READ // '0644'
                         };
                         if (obj.type === 'state') {
-                            // @ts-expect-error we have ensured, that property acl exists
-                            obj.acl.state =
+                            obj.acl!.state =
                                 (this.defaultNewAcl && this.defaultNewAcl.state) ||
                                 CONSTS.ACCESS_USER_RW | CONSTS.ACCESS_GROUP_READ | CONSTS.ACCESS_EVERY_READ; // '0644'
                         }
                     }
                     if (options.object !== undefined) {
                         obj.acl.object = options.object;
                     }
-                    if (options.state !== undefined) {
-                        // @ts-expect-error TODO: state not allowed or types incorrect?
+
+                    if (options.state !== undefined && 'state' in obj.acl) {
```

Comment:
I think the more logical check would be that the object is a StateObject
```suggestion
                    if (obj.type === 'state' && options.state !== undefined) {
```
but I'm not 100% sure that TS gets that.

Reply:
Then it still needs the check like `obj.acl!.state` for which i have added the guard

Reply:
Hm probably the same weird narrowing behavior as above then 🤷‍♂️